sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1765,6 +1765,24 @@ object functions {
   }
 
   /**
+   * Returns string, with the first letter of each word in uppercase,
+   * all other letters in lowercase. Words are delimited by whitespace.
+   *
+   * @group string_funcs
+   * @since 1.5.0
+   */
+  def initcap(e: Column): Column = InitCap(e.expr)
+
+  /**
+   * Returns string, with the first letter of each word in uppercase,
+   * all other letters in lowercase. Words are delimited by whitespace.
+   *
+   * @group string_funcs
+   * @since 1.5.0
+   */
+  def initcap(columnName: String): Column = initcap(Column(columnName))
```

Comment:
As @rxin made some clean up for the DF function, we'd better remove the `columnName` version of API.